### ISSUE TITLE
[C-Extension] Compilation Fails with changes to to macro-assembler

### ISSUE THREAD

**Issue by derekztu22** [v8-riscv/v8#326]
**Describe the bug**
./mksnapshot runs into Out Of Memory error with changes

**To Reproduce**
1. Run and build with code following instructions in [wiki](https://github.com/v8-riscv/v8/wiki/Simulator-Build)
2. See error

**Expected behavior**
Compilations passes

**Errors/Logs**
```
FAILED: gen/embedded.S snapshot_blob.bin
python ../../tools/run.py ./mksnapshot --turbo_instruction_scheduling --target_os=linux --target_arch=x64 --embedded_src gen/embedded.S --target_is_simulator --embedded_variant Default --random-seed 314159265 --startup_blob snapshot_blob.bin --native-code-counters --verify-heap
<--- Last few GCs --->
<--- JS stacktrace --->
#
# Fatal process OOM in Assembler::GrowBuffer
#
Return code is -4
```

**Additional context**
Changes are as below:

`void TurboAssembler::Add64(Register rd, Register rs, const Operand& rt) {
  if (rt.is_reg()) {
    if ((rd.code() == rt.rm().code()) && (rt.rm() != zero_reg)) {
      c_add(rs, rt.rm());
    } else {
      add(rd, rs, rt.rm());
    }
  } else {`

Also see [#285](https://github.com/v8-riscv/v8/issues/285)

**Comment by derekztu22** [comment]
@fw-brice here is the issue for the bug we discussed on Wednesday. Let's have all discussions here.

**Comment by fw-brice** [comment]
A general comment for debugging errors from mksnapshot - you can run mksnapshot in the VSCode debugger by following the configuration [here](https://github.com/v8-riscv/v8/wiki/VSCode-Setup#vscodelaunchjson) or you can extract the command from that to run in gdb.

**Comment by fw-brice** [comment]
I would recommend looking into any places that use `kInstrSize` or `kInstrSizeLog2` and verify that they are correctly handling variable size instructions. The method `InstructionSize()` can alternatively be used on an instruction to retrieve its size. I have not looked closely at any of that code to verify if it is properly modified.

**Comment by fw-brice** [comment]
One other tip is to try to build v8 without the snapshot, so that the build can succeed, then try to find a simpler test case to reproduce the problem. You can disable snapshot by adding `v8_use_snapshot=false` to args.gn in your build directory.

**Comment by derekztu22** [comment]
Has been traced up to src/compiler/backend/code-generator.cc:407
`tasm()->Align(Code::kMetadataAlignment);`

Will continue to dig deeper.

**Comment by derekztu22** [comment]
Fixed see PR #369.
